src/main/java/pl/schibsted/services/DirectoryService.java

```diff
@@ -12,13 +17,25 @@
 
 public class DirectoryService {
 
+	public static final FilenameFilter TEXT_FILES_FILENAME_FILTER = (dir, name) -> name.endsWith(".txt");
+
 	private final Directory directory;
 	private final Set<DirectoryValidator> validators = Set.of(new DirectoryExistsValidator(), new AnyTextFilePresentValidator());
 
 	public DirectoryService(Directory directory) {
 		this.directory = directory;
 	}
 
+	public Set<Path> getTextFiles() {
+		final File[] textFiles = new File(this.directory.getPath()).listFiles(TEXT_FILES_FILENAME_FILTER);
+		if (textFiles != null) {
+			return Arrays.stream(textFiles)
```

Comment:
Możesz zrobić optional.ofnullable(textfiles).stream wtedy tego idą nie potrzeba chyba ze to Java 8 to to co jest na dole możesz zamienić na collections.emptyset()